setup.py

```diff
@@ -22,33 +22,36 @@
 from setuptools import find_packages, setup
 
 INSTALL_REQUIRES = [
-    "Twisted>=19.7",
-    "prometheus_client>=0.7.0,<0.8",
     "aioapns>=1.10",
+    "attrs>=19.2.0",
     "cryptography>=2.6.1",
-    "pyyaml>=5.1.1",
-    "service_identity>=18.1.0",
+    "idna>=2.8",
+    "importlib_metadata",
     "jaeger-client>=4.0.0",
+    "matrix-common==1.0.0",
     "opentracing>=2.2.0",
+    "prometheus_client>=0.7.0,<0.8",
+    "py-vapid>=1.7.0",
+    "pyOpenSSL>=17.5.0",
+    "pywebpush>=1.13.0",
+    "pyyaml>=5.1.1",
     "sentry-sdk>=0.10.2",
+    "service_identity>=18.1.0",
+    "Twisted>=19.7",
+    "typing-extensions>=3.7.4",
```

Comment:
Should we make this conditional so it's only installed on Py < 3.8? (And do the same for the import that needs it)

Reply:
I think we typically import directly from `typing_extensions`; would have to change the source to do a `try...except ImportError...` dance

Reply:
On the import side, mypy does not appreciate the obvious attempt:
https://mypy-play.net/?mypy=latest&python=3.10&gist=9ac99f23480e5c8bc86c68f94ce65b6c

```py
if TYPE_CHECKING or sys.version_info < (3, 8, 0):
    from typing_extensions import Literal
else:
    from typing import Literal
```
works with mypy but always requires typing_extensions in dev.

Reply:
There's a PR to do the same for python-signedjson here: https://github.com/matrix-org/python-signedjson/pull/12